java-source/src/main/java/net/corda/training/state/IOUState.java

```diff
@@ -1,45 +1,48 @@
 package net.corda.training.state;
 
-import com.google.common.collect.ImmutableList;
-
-import java.util.*;
-
 import net.corda.core.contracts.Amount;
+import net.corda.core.contracts.ContractState;
 import net.corda.core.contracts.LinearState;
 import net.corda.core.contracts.UniqueIdentifier;
 import net.corda.core.identity.Party;
 import net.corda.core.identity.AbstractParty;
+
+import java.util.*;
+import com.google.common.collect.ImmutableList;
 import net.corda.core.serialization.ConstructorForDeserialization;
 
+import javax.validation.constraints.NotNull;
+
 /**
  * The IOU State object, with the following properties:
  * - [amount] The amount owed by the [borrower] to the [lender]
  * - [lender] The lending party.
  * - [borrower] The borrowing party.
+ * - [contract] Holds a reference to the [IOUContract]
  * - [paid] Records how much of the [amount] has been paid.
  * - [linearId] A unique id shared by all LinearState states representing the same agreement throughout history within
- * the vaults of all parties. Verify methods should check that one input and one output share the id in a transaction,
- * except at issuance/termination.
+ *   the vaults of all parties. Verify methods should check that one input and one output share the id in a transaction,
+ *   except at issuance/termination.
  */
-public class IOUState implements LinearState {
-    private final Amount<Currency> amount;
-    private final Party lender;
-    private final Party borrower;
-    private final Amount<Currency> paid;
+public class IOUState implements ContractState, LinearState {
+
+    public final Amount<Currency> amount;
+    public final Party lender;
+    public final Party borrower;
+    public final Amount<Currency> paid;
     private final UniqueIdentifier linearId;
 
     // Private constructor used only for copying a State object
     @ConstructorForDeserialization
-    private IOUState(Amount<Currency> amount, Party lender, Party borrower, Amount<Currency> paid, UniqueIdentifier linearId) {
-        this.amount = amount;
-        this.lender = lender;
-        this.borrower = borrower;
-        this.paid = paid;
-        this.linearId = linearId;
-    }
+    public IOUState(Amount<Currency> amount, Party lender, Party borrower, Amount<Currency> paid, UniqueIdentifier linearId){
```

Comment:
This should be private because the app developer should not be able to set paid or linearId